src/main/java/com/adobe/abp/regola/rules/RangeRule.java

```diff
@@ -0,0 +1,304 @@
+ * </ul>
+ *
+ * @param <T> type of the fact (must extend Comparable)
+ */
+@Experimental
+public class RangeRule<T extends Comparable<T>> extends OperatorBasedRule {
+
+    private final Executor executor;
+    private T min;
+    private T max;
+    private T step;
+    private boolean minExclusive;
+    private boolean maxExclusive;
+
+    public RangeRule() {
+        this(null);
+    }
+
+    public RangeRule(Executor executor) {
+        super(RuleType.RANGE.getName());
+        this.executor = executor;
+    }
+
+    public T getMin() {
+        return min;
+    }
+
+    public RangeRule<T> setMin(T min) {
+        this.min = min;
+        return this;
+    }
+
+    public T getMax() {
+        return max;
+    }
+
+    public RangeRule<T> setMax(T max) {
+        this.max = max;
+        return this;
+    }
+
+    public T getStep() {
+        return step;
+    }
+
+    public RangeRule<T> setStep(T step) {
+        this.step = step;
+        return this;
+    }
+
+    public boolean isMinExclusive() {
+        return minExclusive;
+    }
+
+    public RangeRule<T> setMinExclusive(boolean minExclusive) {
+        this.minExclusive = minExclusive;
+        return this;
+    }
+
+    public boolean isMaxExclusive() {
+        return maxExclusive;
+    }
+
+    public RangeRule<T> setMaxExclusive(boolean maxExclusive) {
+        this.maxExclusive = maxExclusive;
+        return this;
+    }
+
+    @SuppressWarnings("unchecked")
+    private T convertToTargetType(Number value) {
+        if (min == null) {
+            throw new IllegalStateException("Min value must be set to determine the target type");
+        }
+
+        if (min instanceof Integer) {
+            return (T) Integer.valueOf(value.intValue());
+        } else if (min instanceof Long) {
+            return (T) Long.valueOf(value.longValue());
+        } else if (min instanceof Double) {
+            return (T) Double.valueOf(value.doubleValue());
+        } else if (min instanceof Float) {
+            return (T) Float.valueOf(value.floatValue());
+        } else if (min instanceof Short) {
+            return (T) Short.valueOf(value.shortValue());
+        } else if (min instanceof Byte) {
+            return (T) Byte.valueOf(value.byteValue());
+        }
+
+        throw new IllegalArgumentException("Unsupported numeric type: " + min.getClass());
+    }
+
+    private Collection<T> generateExpectedValues() {
+        Collection<T> expectedValues = new ArrayList<>();
+
+        switch (getOperator()) {
+            case BETWEEN:
+                expectedValues.add(min);
+                expectedValues.add(max);
+                break;
+            case IN:
+                if (step != null && min instanceof Number && max instanceof Number && step instanceof Number) {
+                    double minVal = ((Number) min).doubleValue();
+                    double maxVal = ((Number) max).doubleValue();
+                    double stepVal = ((Number) step).doubleValue();
```

Comment:
The type should be on the class: RangeRule<T>, so there should be no need for this.